### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by txasw**
This pull request adds a comprehensive set of test data files to cover a wide range of scenarios for system testing data validation. The changes include updates to the main `system_testing_data.csv` with an enhanced schema, and the addition of multiple specialized test files for edge cases, invalid inputs, and data integrity checks.

**Major updates to main test data:**

* Expanded the `system_testing_data.csv` file to include a new `Active` column and updated all records to match the new schema, supporting more robust test scenarios.

**Addition of specialized test data files:**

_Data integrity and edge cases:_
* Added files for all-active (`all_active.csv`), all-inactive (`all_inactive.csv`), all-passed (`all_passed.csv`), all-failed (`all_failed.csv`), all-pending (`all_pending.csv`), and all-success (`all_success.csv`) scenarios, as well as files for exact duplicates (`exact_duplicates.csv`), duplicate IDs (`duplicate_ids.csv`), empty data (`empty_data.csv`), empty fields (`empty_fields.csv`), and extreme values (`extreme_values.csv`). [[1]](diffhunk://#diff-94aca4143ddf8943ccbee4a8a2decdcf2e645bdc2f79cc3fba2f4a0ea5c0a31aR1-R6) [[2]](diffhunk://#diff-de85709ca5a13ebf97a3cd59b5fb4c3be7020283258d18dfb4bb5ae35a12dda6R1-R6) [[3]](diffhunk://#diff-87c3bf9a462590425f84c5332239a4e897e4108f902822ba06ce88918be536b4R1-R6) [[4]](diffhunk://#diff-442fb76f6794b3f87da0aa63639d47aecd31b7e6fa310c5fcbd13fc3a748a9f5R1-R6) [[5]](diffhunk://#diff-1c60170180527090e70e31277a0dd47e0caaa364c2d331005eb2808023cfa539R1-R6) [[6]](diffhunk://#diff-92d9e99dfc6c3a287b40dea9a7d3f22982cab8f32caf0c698b861f1a0d008a6dR1-R6) [[7]](diffhunk://#diff-fa23e32c0ac26ed74c4c2b63c5f9e17e1b45fb032571238e8d30256da32292e4R1-R7) [[8]](diffhunk://#diff-e03c59f8a8c3d1ba15c8639ef47365853f13f48eeb71f6da9b889834e0b66344R1-R6) [[9]](diffhunk://#diff-e6d3ec77826a5384de00a4a5513463378962bfe1a77e43ed50703b51562f5d11R1) [[10]](diffhunk://#diff-afacb598e814f8c56a9eeb7789c52eb6d29dabdca4821e048c77ad0dd53e98e5R1-R7) [[11]](diffhunk://#diff-5834b1669dc49c8bea787c2ce53ba6e3c7bb2a23f6774dfcb5d86af945473b6eR1-R6)

_Input validation:_
* Added files to test invalid test IDs (`invalid_test_ids.csv`), invalid names (`invalid_names.csv`), invalid test types (`invalid_test_types.csv`), invalid test results (`invalid_results.csv`), invalid active values (`invalid_active_values.csv`), and extra fields (`extra_fields.csv`). [[1]](diffhunk://#diff-1f7abc766e0bff609b9f8a2cfa1856a0aa64a4a15894397d1191aff8f1f8b83aR1-R6) [[2]](diffhunk://#diff-f8b2bded038103e8f0b83e6b5c574622a0fc789aea7b7243ea9ccf634a20e170R1-R11) [[3]](diffhunk://#diff-d25c325596da5a3281537486f00964231b6bf0213b384853be3629cd2971f6a3R1-R11) [[4]](diffhunk://#diff-5ebb794ec345b67ac9531e3ea80709031e41950d9cdef7ac442303b92474cca0R1-R11) [[5]](diffhunk://#diff-3070261d9582f7bfe768466bdfa68c18172a6112c8dec70d7091eec2d31b661fR1-R11) [[6]](diffhunk://#diff-17244141db98b84f549ca3695e95ea5689faff00afb04387d0bdcd971e2a6977R1-R3)

_Schema and format validation:_
* Added a file with a wrong header and invalid format (`invalid_test.csv`).
* Added a file with all possible combinations of test types and results for thorough validation (`all_combinations.csv`).

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -1,103 +1,2212 @@
...
+#endif
+
+// Main function
+int main(void)
+{
+    pause_screen();
+    clear_screen();
+    printf("╔══════════════════════════════════════════════════════════════╗\n");
+    printf("║                 SYSTEM TESTING DATA MANAGER                  ║\n");
+    printf("║                  ระบบจัดการข้อมูลกรทดสอบระบบ                    ║\n");
```
[the remark this answers is not in the document]
Comment (txasw) [reply main.c:2133]: I intend to make the display beautiful
